src/openrct2/ride/coaster/LoopingRollerCoaster.cpp

```diff
@@ -31,33 +31,9 @@ static void LoopingRCTrackFlat(
 {
     if (trackElement.HasChain())
     {
-        switch (direction)
-        {
-            case 0:
-                PaintAddImageAsParentRotated(
-                    session, direction, session.TrackColours[SCHEME_TRACK].WithIndex(15006), { 0, 0, height }, { 32, 20, 3 },
-                    { 0, 6, height });
-                break;
-            case 1:
-                PaintAddImageAsParentRotated(
-                    session, direction, session.TrackColours[SCHEME_TRACK].WithIndex(15007), { 0, 0, height }, { 32, 20, 3 },
-                    { 0, 6, height });
-                break;
-            case 2:
-                PaintAddImageAsParentRotated(
-                    session, direction, session.TrackColours[SCHEME_TRACK].WithIndex(15008), { 0, 0, height }, { 32, 20, 3 },
-                    { 0, 6, height });
-                break;
-            case 3:
-                PaintAddImageAsParentRotated(
-                    session, direction, session.TrackColours[SCHEME_TRACK].WithIndex(15009), { 0, 0, height }, { 32, 20, 3 },
-                    { 0, 6, height });
-                break;
-        }
-        if (TrackPaintUtilShouldPaintSupports(session.MapPosition))
-        {
-            MetalASupportsPaintSetup(session, METAL_SUPPORTS_TUBES, 4, 0, height, session.TrackColours[SCHEME_SUPPORTS]);
-        }
+        PaintAddImageAsParentRotated(
+            session, direction, session.TrackColours[SCHEME_TRACK].WithIndex(15006 + direction), { 0, 0, height },
```

Comment:
Use the call with BoundBoxXYZ

Reply:
Could you give me an example for this for reference?

Reply:
Look at PaintAddImageAsParent, there is a function overload that uses BoundBoxXYZ.

Reply:
As mentioned in my other PR, I think it's a bit beyond the scope of this specific PR. Would be better for a separate.